pkg/model/pki.go

```diff
@@ -30,65 +32,80 @@ type PKIModelBuilder struct {
 
 var _ fi.ModelBuilder = &PKIModelBuilder{}
 
+// Build is responsible for generating the pki assets for the cluster
 func (b *PKIModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	{
-		// Keypair used by the kubelet
 		t := &fitasks.Keypair{
 			Name:      fi.String("kubelet"),
 			Lifecycle: b.Lifecycle,
-
-			Subject: "o=" + user.NodesGroup + ",cn=kubelet",
-			Type:    "client",
+			Subject:   "o=" + user.NodesGroup + ",cn=kubelet",
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
 	{
-		// Secret used by the kubelet
-		// TODO: Can this be removed... at least from 1.6 on?
-		t := &fitasks.Secret{
-			Name:      fi.String("kubelet"),
+		t := &fitasks.Keypair{
+			Name:      fi.String("kube-scheduler"),
 			Lifecycle: b.Lifecycle,
+			Subject:   "cn=" + user.KubeScheduler,
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
 	{
-		// Keypair used by the kube-scheduler
 		t := &fitasks.Keypair{
-			Name:      fi.String("kube-scheduler"),
+			Name:      fi.String("kube-proxy"),
 			Lifecycle: b.Lifecycle,
-
-			Subject: "cn=" + user.KubeScheduler,
-			Type:    "client",
+			Subject:   "cn=" + user.KubeProxy,
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
 	{
-		// Secret used by the kube-scheduler
-		// TODO: Can this be removed... at least from 1.6 on?
-		t := &fitasks.Secret{
-			Name:      fi.String("system:scheduler"),
+		t := &fitasks.Keypair{
+			Name:      fi.String("kube-controller-manager"),
 			Lifecycle: b.Lifecycle,
+			Subject:   "cn=" + user.KubeControllerManager,
+			Type:      "client",
 		}
 		c.AddTask(t)
 	}
 
-	{
-		// Keypair used by the kube-proxy
-		t := &fitasks.Keypair{
-			Name:      fi.String("kube-proxy"),
-			Lifecycle: b.Lifecycle,
-
-			Subject: "cn=" + user.KubeProxy,
-			Type:    "client",
+	// @check if we need to generate certificates for etcd peers certificates from a different CA
+	// @question i think we should use another KeyStore for this, perhaps registering a EtcdKeyStore given
+	// that mutual tls used to verify between the peers we don't was
```

Comment:
Not sure I understand this comment in full.

I think a separate etcd CA is a good idea, but I don't know if we need to do it right away.  Because this feature is effectively gated behind `EnableEtcdTLS` I'm inclined to say we merge and iterate.  A concern is that by sharing the same etcd CA key and k8s CA key, any k8s certificate will permit access to etcd, negating some of the security benefits of etcd TLS. 

One option is that we could have a single self-signed CA root, but then have it sign intermediate certs, one for etcd, one for API etc.  In practice it may be easier to just have independent CAs though..